src/main/java/com/google/api/generator/spring/composer/SpringAutoConfigClassComposer.java

```diff
@@ -659,10 +659,13 @@ private static MethodDefinition createSettingsBeanMethod(
     List<Statement> updateRetrySettingsStatementBody = new ArrayList<>();
 
     for (Method method : service.methods()) {
-      List<Statement> updateMethodWithServiceRetryStatments =
+      if (!method.stream().equals(Method.Stream.NONE) || method.hasLro()) {
```

Comment:
I think it would make sense to filter the methods in the foreach declaration (e.g. `for (Method method : Utils.getFilteredMethods(service.methods))` to centralize/dereplicate the filtering logic (could also be done by adding a convenience method in `Service` but may be overkill)

Reply:
Ah, I missed this was a draft. Hopefully it was still a useful catch!

Reply:
Makes sense, given we expect this to be changed in the future. Updated and moved this logic to utils.